keylime/da/examples/file.py

```diff
@@ -0,0 +1,121 @@
+import os
+
+from keylime import keylime_logging
+from keylime.da.record import BaseRecordManagement, build_key_list, mkdir_p
+
+# setup logging
+logger = keylime_logging.init_logging("durable_attestation_persistent_store")
+
+# ######################################################
+# Durable Attestation record manager with "plain file" backend
+# ######################################################
+
+
+class RecordManagement(BaseRecordManagement):
+    def __init__(self, service):
+
+        BaseRecordManagement.__init__(self, service)
+        self.rcd_enc = "base64"
+        self.file_path = self.ps_url.path
+        self.file_prefix = self.ps_url.query.replace("prefix=", "")
+        self.line_sep = b"\n----\n"
+
+        mkdir_p(self.file_path)
+
+    def agent_list_retrieval(self, record_prefix="auto", service="auto"):
+
+        if record_prefix == "auto":
+            record_prefix = self.file_prefix
+
+        _agent_list = []
+
+        record_prefix += "_" + service
+        logger.debug(
+            "Extracting the UUIDs of all agents with entries with prefix %s from filesystem persistent store",
+            record_prefix,
+        )
+        for _entry in next(os.walk(self.file_path), (None, None, []))[2]:
+            if service in _entry:
+                _agent_uuid = _entry.replace(record_prefix + "_", "").replace("." + self.rcd_fmt, "")
+                if _agent_uuid not in _agent_list:
+                    _agent_list.append(_agent_uuid)
+
+        return _agent_list
+
+    def bulk_record_retrieval(self, record_identifier, record_list):
+
+        logger.debug(
+            "Extracting all records for record_identifier %s from filesystem persistent store", record_identifier
+        )
+
+        with open(record_identifier, "rb") as fp:
+            for _line in fp:
+                if b"\n" + _line != self.line_sep:
+                    _encoded_record_object = _line
+
+                    _decoded_record_object = self._record_deserialize(_encoded_record_object)
```

Comment:
See, there is an access here of an inherited method that start with `_`.

IMHO drop all the `_` everywhere, including variables. There is a meaning in `_` and `__` in Python, but is mostly a convention. If this is not followed consistently creates confusion : ))

Reply:
done